airflow/providers/google/cloud/triggers/bigquery.py

```diff
@@ -89,6 +90,19 @@ def serialize(self) -> tuple[str, dict[str, Any]]:
             },
         )
 
+    def safe_to_cancel(self) -> bool:
+        """
+        Whether it is safe to cancel the external job which is being executed by this trigger.
+
+        This is to avoid the case that `asyncio.CancelledError` is called because the trigger itself is stopped.
+        Because in those cases, we should NOT cancel the external job.
+        """
+        self.log.info("Checking if it is safe to cancel the job.")
+        return self.task_instance not in {
+            TaskInstanceState.RUNNING,
+            TaskInstanceState.DEFERRED,
+        }
```

Comment:
What is the value for taskinstance when task is marked failed externally?

Reply:
Its deferred

Reply:
Isn't it safe to cancel only if it's in deferred state then?